### PR TITLE
Fixes #20463: Add setup custom port (for https and cfengine)

### PR DESCRIPTION
**Pull request by julienbriault**
https://issues.rudder.io/issues/20463

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,6 +139,8 @@ and installs the packages.
 - `rudder_repository`: Rudder repository domain (default: `repository.rudder.io`)
 - `rudder_repository_username`: Optional username to pass to repository if using credentials
 - `rudder_repository_password`: Optional password to pass to repository if using credentials
+- `rudder_policy_distribution_cfengine_port`: Optional custom port for Cfengine (default: `5309`)
```
Comment (amousset) [inline README.md:142]:
```suggestion
- `rudder_policy_distribution_cfengine_port`: Optional custom port for CFEngine (default: `5309`)
```

Comment (julienbriault) [reply README.md]: Fixup !

File: roles/rudder_agent/tasks/configure.yml [hunk 2]
```diff
@@ -5,3 +5,21 @@
...
+    replace: "{{ rudder_policy_distribution_cfengine_port }}"
+  when: "{{ agent_version is version('6', operator='lt', strict=False) }}"
+
+- name: Set custom Rudder ports for 7.0 (min) version
+  replace:
+    path: "{{ item.path }}"
+    regexp: "{{ item.regex }}"
+    replace: "{{ item.replace }}"
+  with_items:
+    - { path: "/etc/rudder/etc/agent.conf", regex: "^https_port=(.*)$", replace: "{{ rudder_policy_distribution_https_port }}" },
```
Comment (amousset) [inline roles/rudder_agent/tasks/configure.yml:22]: does this create the file and add the line if not present?

Comment (julienbriault) [reply roles/rudder_agent/tasks/configure.yml:22]: this function only allows to check and replace a value in a file, I'm not sure which one creates the file if it doesn't exist

Comment (amousset) [reply roles/rudder_agent/tasks/configure.yml:22]: We need to create the file as it does not exist by default

File: roles/rudder_agent/tasks/configure.yml [hunk 3]
```diff
@@ -5,3 +5,21 @@
     owner: root
     group: root
     mode: '0644'
+
+- name: Set custom Rudder ports for 6 (min) version
+  replace:
```
Comment (amousset) [inline roles/rudder_agent/tasks/configure.yml:10]: won't this conflict with the file content just above?

File: roles/rudder_agent/tasks/configure.yml (hunk 2 above)
Comment (julienbriault) [reply roles/rudder_agent/tasks/configure.yml:22]: it doesn't seem to be in conflict, in any case, no problem during my tests.

File: roles/rudder_agent/tasks/configure.yml (hunk 3 above)
Comment (julienbriault) [reply roles/rudder_agent/tasks/configure.yml:10]: it doesn't seem to be in conflict, in any case, no problem during my tests.

File: roles/rudder_agent/tasks/configure.yml (hunk 2 above)
Comment (julienbriault) [reply roles/rudder_agent/tasks/configure.yml:22]: Ok, I fixup the pr with the file creation!

File: roles/rudder_agent/tasks/configure.yml (hunk 3 above)
Comment (amousset) [reply roles/rudder_agent/tasks/configure.yml:19]: doesn't it make 2 repairs every time it runs? (on to reset the content of the file to the policy server hostname and one to set the port?

File: roles/rudder_agent/tasks/configure.yml [hunk 4]
```diff
@@ -1,7 +1,34 @@
...
+    state: touch
+    owner: root
+    mode: 0644
+    modification_time: preserve
+    access_time: preserve
+
+- name: Set custom Rudder ports for 6 (min) version
+  replace:
+    path: "{{ policy_server_dat_file }}"
+    regexp: "^x=rudder.example.com:(.*)$"
```
Comment (amousset) [inline roles/rudder_agent/tasks/configure.yml:21]: I don't get the `x=` here, and no the `rudder.example.com` either, shouldn't this match the currently configured policy server name?

Comment (julienbriault) [reply roles/rudder_agent/tasks/configure.yml:21]: Indeed, I had simply based myself on what is referenced in the documentation.

Comment (julienbriault) [reply roles/rudder_agent/tasks/configure.yml:21]: Doc: https://docs.rudder.io/reference/6.2/administration/port.html#_set_connection_port_on_the_agent

**Comment by julienbriault** [comment]
Everything has been corrected.

**Comment by amousset** [comment]
Rebase required

**Pull request closed without merging**